### DISCUSSION TITLE
left over config entries after branch deletion

### DISCUSSION THREAD

**Discussion by extrawurst** [libgit2/libgit2#6190] [Q&A]
given a local branch tracking a remote (`branch-c` here) the config contains something like this:

```
[branch "branch-a"]
[branch "branch-b"]
[branch "branch-c"]
        remote = origin
        merge = refs/heads/branch-c
```

after deleting it via `git_branch_delete` only the remote config will disappear:
```
[branch "branch-a"]
[branch "branch-b"]
[branch "branch-c"]
```

this accumulates a bunch of trash in the config.

Am I deleting branches the wrong way? or is this a bug?

**Comment by ethomson** [comment]
It is a bug - but it should have no impact beyond aesthetics.  (The config produced is still technically correct, even if there's cruft.)  So it hasn't really bubbled up to the top of the priority list.

**Reply by carlosmn** [reply]
#5901 is the bug asking for it
